app/controllers/cities_controller.rb

```diff
@@ -22,7 +22,12 @@ def forbes_index
   end
 
   def forbes_show
-    use_new_styles
+    if current_user
+      use_new_styles
+    else
+      flash[:notice] = 'You should sign up first!'
```

Comment:
not going to care about copy here since the flash messages aren't showing up. probably missing from the template, but don't really care.